File: Euclid/Messages/Outgoing/USEROBJECT.cs
using Euclid.Network.Streams.Util;
using Euclid.Storage.Database.Data;

namespace Euclid.Messages.Outgoing
{
    class USEROBJECT : IMessageComposer
    {
        private PlayerData playerData;

        public USEROBJECT(PlayerData playerData)
        {
            this.playerData = playerData;
        }

        public override void Write()
        {
            this.Data.Add(new KeyValueEntry("name", this.playerData.Name, "="));
            this.Data.Add(new KeyValueEntry("figure", this.playerData.Figure, "="));
            this.Data.Add(new KeyValueEntry("email", this.playerData.Email, "="));
            this.Data.Add(new KeyValueEntry("birthday", this.playerData.Birthday, "="));
            this.Data.Add(new KeyValueEntry("phonenumber", this.playerData.PhoneNumber, "="));
            this.Data.Add(new KeyValueEntry("customData", this.playerData.CustomData, "="));
            this.Data.Add(new KeyValueEntry("has_read_agreement", "1", "="));
            this.Data.Add(new KeyValueEntry("sex", this.playerData.Sex, "="));
            this.Data.Add(new KeyValueEntry("country", this.playerData.Country, "="));
            this.Data.Add(new KeyValueEntry("has_special_rights", "1", "="));
            this.Data.Add(new KeyValueEntry("badge_type", "", "="));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/USEROBJECT.java
package euclid.messages.outgoing;

import euclid.messages.IMessageComposer;
import euclid.network.streams.util.KeyValueEntry;
import euclid.storage.database.data.PlayerData;

public class USEROBJECT extends IMessageComposer {

    private final PlayerData playerData;

    public USEROBJECT(PlayerData playerData) {
        this.playerData = playerData;
    }

    @Override
    public void write() {
        getData().add(new KeyValueEntry("name", playerData.getName(), "="));
        getData().add(new KeyValueEntry("figure", playerData.getFigure(), "="));
        getData().add(new KeyValueEntry("email", playerData.getEmail(), "="));
        getData().add(new KeyValueEntry("birthday", playerData.getBirthday(), "="));
        getData().add(new KeyValueEntry("phonenumber", playerData.getPhoneNumber(), "="));
        getData().add(new KeyValueEntry("customData", playerData.getCustomData(), "="));
        getData().add(new KeyValueEntry("has_read_agreement", "1", "="));
        getData().add(new KeyValueEntry("sex", playerData.getSex(), "="));
        getData().add(new KeyValueEntry("country", playerData.getCountry(), "="));
        getData().add(new KeyValueEntry("has_special_rights", "1", "="));
        getData().add(new KeyValueEntry("badge_type", "", "="));
    }
}
